modelscope_agent/memory/base.py

```diff
@@ -0,0 +1,65 @@
+import os
+from typing import Dict, Iterable, List, Union
+
+import json
+from modelscope_agent.schemas import AgentHolder, Message
+from pydantic import ConfigDict
+
+
+class Memory(AgentHolder):
+    path: str
+    model_config = ConfigDict(extra='allow')
+
+    def save_memory(self, history: List[Message]):
+        """
+        save history memory to path
+        Args:
+            history: List of Message
+
+        Returns: None
+
+        """
+        directory = os.path.dirname(self.path)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        with open(self.path, 'w', encoding='utf-8') as file:
```

Comment:
looks weird,  save_memory does not save self.history?   save_memory  load memory. get_history. update_history. seems unrelated between memory and history

Reply:
has changed to reasonable name